### DISCUSSION TITLE
NaN in thumbnail display

### DISCUSSION THREAD

**Discussion by camdendotlol** [ProjectMirador/mirador#3532] [Q&A] [answered]
Hello,

I've set up a Mirador 3 site and, while everything else works fine, I'm seeing these NaN labels on the thumbnails.

I'm assuming it's an issue with our manifest, like a missing or invalid field. Looking through the Mirador code, I can't find where specifically it's looking for this label. Which data field does Mirador source these labels from?

![Screen Shot 2022-02-02 at 10 24 09 AM](https://user-images.githubusercontent.com/64725469/152183676-c29a4f95-4907-4797-a988-cec5468c9b3c.png)

**Comment by northcoder-repo** [comment, marked as answer]
I just encountered this issue. Here is how I resolved it, using IIIF presentation API v3.0:

I added the following `label` section to my items list, where `"type": "Canvas"`:

```
"label": {
    "en": [
        "foo"
    ]
},
```
Now I see `foo` in the thumbnail, instead of `NaN` ("not a number"). 

(This was just for a single image. I would repeat this for multiple images, as needed. Haven't got to that step yet. I am new to IIIF, so there is always the chance that there is a better approach I am not aware of.)

---

If you are using version 2 of the presentation spec, I think the solution is completely different - relating to [`sequences`](https://iiif.io/api/presentation/2.1/#sequence) labels.

**Comment by enriquediaz** [comment]
@camdendotlol would you be able to provide a copy of a manifest where you're seeing this occur? 

While I think @northcoder-repo has indeed come up with a solution that highlights a bug in Mirador where it attempts to construct a label when no label is given (credit to @lutzhelm for spotting [where in the code this is occurring](https://github.com/ProjectMirador/mirador/blob/master/src/components/IIIFThumbnail.js#L126-L139)), it's also possible the manifest design can also be contributing to this.

**Reply by camdendotlol** [reply]
When I get back to work on Monday I can try to find the manifests we used for that project but I'll have to ask around a bit to see if it's alright for me to share one of the manifests here.

I do remember that we fixed the issue by adding labels like northcoder-repo did.

**Reply by enriquediaz** [reply]
In that case I'll close this and if we do come across an issue again, we'll hopefully be able to respond more quickly!
